### PR TITLE
fix etcdv3 client watch: reuse previous watch to avoid memory leak

### LINKED ISSUE #615: Memory leaks with etcdv3 mode confd v0.14.0 (13 too)
Runs like:
`./confd-0.14.0-linux-amd64 -backend etcdv3 -srv-record _etcd-client-ssl._tcp.dev.lan -client-ca-keys /etc/confd/certs/ca.pem -client-cert /etc/confd/certs/etcd-client.pem -client-key=/etc/confd/certs/etcd-client-key.pem -watch`

Memory stat:
```
# while true; do ps -eo pid,vsz,rss,cmd,time -q `pidof confd-0.14.0-linux-amd64`; sleep 60; done
  PID    VSZ   RSS CMD                             TIME
25809  27064 16024 ./confd-0.14.0-linux-amd64  00:00:00
  PID    VSZ   RSS CMD                             TIME
25809  45800 34420 ./confd-0.14.0-linux-amd64  00:00:01
  PID    VSZ   RSS CMD                             TIME
25809  89328 78388 ./confd-0.14.0-linux-amd64  00:00:03
  PID    VSZ   RSS CMD                             TIME
25809 157232 145556 ./confd-0.14.0-linux-amd64  00:00:05
  PID    VSZ   RSS CMD                             TIME
25809 260424 249392 ./confd-0.14.0-linux-amd64  00:00:07
```

```
./confd-0.14.0-linux-amd64 -version
confd 0.14.0 (Git SHA: 9fab9634, Go Version: go1.9.1)

etcd --version
etcd Version: 3.1.7
Git SHA: 43b7507
Go Version: go1.7.4
Go OS/Arch: linux/amd64
```

### PR DESCRIPTION
**Pull request by zyf0330**
@okushchenko Please check code, I have no too much experience on Go.

Closes #615

### CODE REVIEW

File: backends/etcdv3/client.go [hunk 1]
```diff
@@ -11,16 +11,22 @@ import (
...
 	"github.com/coreos/etcd/clientv3"
+	"sync"
 )
 
 // Client is a wrapper around the etcd client
 type Client struct {
-	client *clientv3.Client
+	client  *clientv3.Client
+	watches map[string]clientv3.WatchChan
+	mu      sync.Mutex
```
Comment (okushchenko) [inline backends/etcdv3/client.go]: The `WatchPrefix` is called synchronously (https://github.com/kelseyhightower/confd/blob/20b3d37da7aaa2c176c0612202c06c5ba4f7d987/resource/template/processor.go#L96), so there is no need to create a mutex. Remove it.

**Comment by zyf0330** [comment]
Do I need re-create this PR?

**Comment by zyf0330** [comment]
I have remove `sync.Mutex`

**Comment by zyf0330** [comment]
This commit doesn't fix **feign death** problem. So I remove one *Close*

**Pull request merged**